src/xSTIR/cSTIR/include/sirf/STIR/stir_x.h

```diff
@@ -107,6 +108,27 @@ namespace sirf {
 		//shared_ptr<stir::ChainedBinNormalisation> norm_;
 	};
 
+	class PoissonNoiseGenerator {
```

Comment:
I'm not sure why you need the SIRF class, but if you do, it needs doxygen comments. I'd also vastly prefer that it follows that Python structure.

Reply:
done

Reply:
well... it doesn't really. There's no `process()` here etc.

anyway.